core/src/main/java/com/yahoo/imapnio/async/client/ImapAsyncClient.java

```diff
@@ -210,16 +214,16 @@ public void operationComplete(final Future<? super Void> future) {
                     }
                     final int sessionId = sessionCount.incrementAndGet();
                     pipeline.addLast(ImapClientConnectHandler.HANDLER_NAME,
-                            new ImapClientConnectHandler(sessionFuture, LoggerFactory.getLogger(ImapAsyncSessionImpl.class), sessionId));
+                            new ImapClientConnectHandler(sessionFuture, LoggerFactory.getLogger(ImapAsyncSessionImpl.class), logOpt, sessionId));
 
-                    if (logger.isDebugEnabled()) {
+                    if (isSessionDebugOn || logger.isDebugEnabled()) {
```

Comment:
change to if (isSessionDebugOn || logger.isTraceEnabled)

Reply:
Done